cli/providers/aws/elb/wait.go

```diff
@@ -9,43 +9,70 @@ import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/pchico83/i2kit/cli/schemas/service"
 )
 
-//Wait waits for the instances in a ELB to be registered
-func Wait(name string, config *aws.Config, log *logger.Logger) error {
+//Wait waits for the instances in a ELB to be healthy
+func Wait(s *service.Service, name string, config *aws.Config, log *logger.Logger) error {
+	registeredInstances, err := waitForRegistered(s, name, config, log)
+	if err != nil {
+		return err
+	}
+	log.Printf("Waiting for %s to be healthy in the '%s' load balancer...", registeredInstances, name)
+	for _, instanceID := range registeredInstances {
+		if err := waitForHealthy(name, instanceID, config, log); err != nil {
+			return err
+		}
+	}
+	log.Printf("%v are healthy in the '%s' load balancer.", registeredInstances, name)
+	return nil
+}
+
+func waitForRegistered(s *service.Service, name string, config *aws.Config, log *logger.Logger) ([]string, error) {
 	log.Printf("Waiting for instances to be registered in the '%s' load balancer...", name)
+	retries := 0
+	registeredInstances := []string{}
+	for retries < 30 {
+		registeredInstances = Instances(name, config)
+		log.Printf("%v instances registered in the '%s' load balancer.", registeredInstances, name)
+		if len(registeredInstances) == s.Replicas {
+			break
+		}
+		time.Sleep(10 * time.Second)
+	}
+	if retries == 20 {
+		return nil, fmt.Errorf("Instances do not register in the load balancer after 200 seconds")
+	}
+	return registeredInstances, nil
+}
+
+func waitForHealthy(name, instanceID string, config *aws.Config, log *logger.Logger) error {
 	svc := elb.New(session.New(), config)
 	dii := &elb.DescribeInstanceHealthInput{
 		LoadBalancerName: aws.String(name),
 	}
-	registered := 0
-	current := 0
-	outOfService := true
-	limit := 60 * 20
-	for outOfService && limit >= 0 {
-		time.Sleep(5 * time.Second)
-		limit -= 5
+	retries := 0
+	startTime := new(int64)
+	*startTime = 0
+	healthy := false
+	for !healthy && retries < 90 {
```

Comment:
Same as above. I think it would simplify the code to just return  on line 65 instead of checking for the value of healthy and max retries to determine success/error